compiler/noirc_evaluator/src/ssa/opt/constant_folding.rs

```diff
@@ -78,25 +94,38 @@ impl Context {
 
         // Cache of instructions without any side-effects along with their outputs.
         let mut cached_instruction_results: HashMap<Instruction, Vec<ValueId>> = HashMap::default();
+        let mut constrained_values: HashMap<ValueId, HashMap<ValueId, ValueId>> =
+            HashMap::default();
```

Comment:
Can you leave a comment to specify what this maps? E.g. `Maps side_effects_var -> constrained value -> simplified value`

Reply:
Good point. I'll do a pass of documentation in general.